xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-resource/resourcePicker.js

```diff
@@ -35,7 +36,9 @@ define('resourcePicker', [
           '<button type="button" class="resourceType btn btn-default">' +
             '<span class="icon">' +
           '</button>' +
-          '<button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown">' +
+          '<button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" ' +
+            'title="' + translations.get('dropdown.toggle.title') + '">' +
```

Comment:
is the result of `translation.get` html safe or does it needs an xml escaping?

Reply:
https://github.com/search?q=repo%3Axwiki%2Fxwiki-platform%20translations.get(&type=code

There's no other use where we just use it as plain HTML. I'm pretty sure using it in `Element.text` escapes it.
From what I can see at https://github.com/xwiki/xwiki-platform/blob/76b653058d9b9fc1c7b2f53f425724a46ed546ea/xwiki-platform-core/xwiki-platform-localization/xwiki-platform-localization-webjar/src/main/webjar/l10n.js#L52-L63

the translation function doesn't care about escaping (which is correct as far as I know).
I couldn't find a proper way to escape things directly in here (except implementing a barebones HTML escaper ^^').
I decided to fill up the template with the translations using jquery methods. These methods escape their inputs so it's safe.

See https://github.com/xwiki/xwiki-platform/pull/3612/commits/edc6d87a475f5dd807727c4a1dba8b687e1840e8

Here is what a quick test of the updated PR gave me (I didn't properly move the translations to my instance - which is why it's somewhat broken in the screenshot.):
![image](https://github.com/user-attachments/assets/9bce08f8-d04b-412c-8396-d6d35d85907a)